asistente_ladm_col/utils/geometry.py

```diff
@@ -238,35 +238,45 @@ def get_overlapping_lines(self, line_layer, use_selection=True):
 
         return dict_res
 
-    def get_overlapping_polygons(self, polygon_layer, id_field=ID_FIELD):
+    def get_overlapping_polygons(self, polygon_layer):
         """
         Obtains overlapping polygons from a single layer
         :param polygon_layer: vector layer with geometry type polygon
-        :param id_field: layer's identifier field
-        :return: List of lists with pairs of overlapping polygons' ids,
+        :return: List of lists with pairs of unique overlapping polygons' ids,
         e.g., [[1, 2], [1, 3]]
         """
         list_overlapping_polygons = list()
-
-        if (QgsWkbTypes.PolygonGeometry != polygon_layer.geometryType() or \
-            polygon_layer.featureCount() == 0):
+        if (QgsWkbTypes.PolygonGeometry != polygon_layer.geometryType()):
             return list_overlapping_polygons
 
-        single_parts_layer = processing.run("native:multiparttosingleparts",
-                                            {'INPUT': polygon_layer, 'OUTPUT': 'memory:'})['OUTPUT']
-        index = QgsSpatialIndex(single_parts_layer)
-
-        for feature in single_parts_layer.getFeatures():
-            bbox = feature.geometry().boundingBox()
-            bbox.scale(1.001)
-            candidates_ids = index.intersects(bbox)
-            candidates_features = single_parts_layer.getFeatures(candidates_ids)
+        features = [i for i in polygon_layer.getFeatures()]
+        features_1 = features_2 = features
 
-            for candidate_feature in candidates_features:
-                is_overlap = feature.geometry().overlaps(candidate_feature.geometry())
-                if is_overlap == True:
-                    overlapping_polygons = sorted([feature[id_field], candidate_feature[id_field]])
-                    if overlapping_polygons not in list_overlapping_polygons:
-                        list_overlapping_polygons.append(overlapping_polygons)
+        for f1 in features_1[:-1]:
+            for f2 in features_2[1:]:
+                isOverlap = f1.geometry().overlaps(f2.geometry())
+                if isOverlap == True:
+                    list_overlapping_polygons.append([f1.id(),f2.id()])
+            del features_2[0] # move the list
 
         return list_overlapping_polygons
+
+    def get_intersection_polygons(self, polygon_layer, polygon_id, overlapping_id):
+        feature_polygon = [feature for feature in polygon_layer.getFeatures() if feature.id() == polygon_id][0]
+        feature_overlap = [feature for feature in polygon_layer.getFeatures() if feature.id() == overlapping_id][0]
+
+        listGeoms = list()
+        intersection = feature_polygon.geometry().intersection(feature_overlap.geometry())
+
+        if (intersection.wkbType() == QgsWkbTypes.Polygon):
+            listGeoms.append(intersection)
+        elif (intersection.wkbType() == QgsWkbTypes.MultiPolygon):
+            listGeoms.append(intersection)
```

Comment:
`if intersection.wkbType() == QgsWkbTypes.Polygon or intersection.wkbType() == QgsWkbTypes.MultiPolygon:`

Reply:
Great!

Reply:
Or if you like:

`if intersection.wkbType() in [QgsWkbTypes.Polygon, QgsWkbTypes.MultiPolygon]:`

However, the recommended way that covers all cases like Polygon, PolygonMZ, MultiPolygon, MultiPolygonZ, etc. is:

`if intersection.type() == QgsWkbTypes.PolygonGeometry:`